LLama/NativeLibraryConfig.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+
+namespace LLama
+{
+#if NET6_0_OR_GREATER
+    /// <summary>
+    /// A class about configurations when loading native libraries. 
+    /// Note that it could be configured only once before any call to llama model apis.
+    /// </summary>
+    public class NativeLibraryConfig
+    {
+        private static NativeLibraryConfig? instance;
+        private static readonly object lockObject = new object();
+
+        /// <summary>
+        /// Whether there's already a config for native library.
+        /// </summary>
+        public bool Initialied { get; private set; }
+        internal Description Desc { get; private set; }
+
+        internal static NativeLibraryConfig GetInstance()
+        {
+            if (instance is null)
+            {
+                lock (lockObject)
+                {
+                    if (instance is null)
+                    {
+                        instance = new NativeLibraryConfig();
+                    }
+                }
+            }
+            return instance;
+        }
+
+        /// <summary>
+        /// Load a specified native library as backend for LLamaSharp
+        /// </summary>
+        /// <param name="libraryPath"></param>
+        /// <exception cref="InvalidOperationException"></exception>
+        public static void WithLibrary(string libraryPath)
+        {
+            var config = GetInstance();
+            if (config.Initialied)
+            {
+                throw new InvalidOperationException("NativeLibraryConfig could be configured only once before any call to llama model apis.");
+            }
+            config.Desc = new Description(libraryPath);
+        }
+
+        /// <summary>
+        /// Ass rules to match a suitable library from installed LLamaSharp backend.
+        /// </summary>
+        /// <param name="useCuda"></param>
+        /// <param name="avxLevel"></param>
+        /// <param name="allowFallback">Whether to allow fall-back when your hardware doesn't support your configuration.</param>
+        /// <param name="skipCheck">Whether to skip the check when fallback is allowed. 
```

Comment:
I'm not really sure what `skipCheck` does from this description. it's "skipping the check", but what check?

Reply:
It's to check if the hardware support the user's choice. Let's ignore the `skipCheck` parameter first, then there're two basic conditions, allowing fallback and the opposite.
If fallback is allowed, we could choose another library if the user's configuration does not match the hardware. For example, the user specified cuda but there's actually no supported cuda version at that machine.

If fallback is disabled, however, we're supposed to load exactly the library specified. However the real condition may be complex and our logic may not cover all the cases (for example, linux cuda detection). Therefore if sometimes loading a library won't actually have problem but we take it as invalid by mistake, the user could force-load it without any check. 

I think by this way we could leave a solution for some potential problems in the future.